packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -0,0 +1,157 @@
+  let gen1Resource = await getResourceDetails('AWS::Cognito::UserPool', gen1UserPoolId, gen1Region);
+  gen1Resource = removeNestedJsonKeys(gen1Resource, ['ProviderURL', 'ProviderName', 'UserPoolId', 'Arn']);
+  // TODO: remove below line after EmailMessage, EmailSubject, SmsMessage, SmsVerificationMessage, EmailVerificationMessage, EmailVerificationSubject, AccountRecoverySetting inconsistency is fixed
+  gen1Resource = removeNestedJsonKeys(gen1Resource, [
+    'UserPoolTags',
+    'VerificationMessageTemplate.EmailMessage',
+    'VerificationMessageTemplate.EmailSubject',
+    'EmailVerificationSubject',
+    'AccountRecoverySetting',
+    'EmailVerificationMessage',
+  ]);
+  const gen2Meta = getProjectOutputs(projRoot);
+  const gen2UserPoolId = gen2Meta.auth.user_pool_id;
+  const gen2Region = gen2Meta.auth.aws_region;
+  let gen2Resource = await getResourceDetails('AWS::Cognito::UserPool', gen2UserPoolId, gen2Region);
+  gen2Resource = removeNestedJsonKeys(gen2Resource, ['ProviderURL', 'ProviderName', 'UserPoolId', 'Arn']);
+  // TODO: remove below line after EmailMessage, EmailSubject, SmsMessage, SmsVerificationMessage, EmailVerificationMessage, EmailVerificationSubject, AccountRecoverySetting inconsistency is fixed
+  gen2Resource = removeNestedJsonKeys(gen2Resource, [
+    'UserPoolTags',
+    'VerificationMessageTemplate.EmailMessage',
+    'VerificationMessageTemplate.SmsMessage',
+    'VerificationMessageTemplate.EmailSubject',
+    'SmsVerificationMessage',
+    'EmailVerificationSubject',
+    'AccountRecoverySetting',
+    'EmailVerificationMessage',
+  ]);
+  expect(gen2Resource).toEqual(gen1Resource);
+}
+
+function removeNestedJsonKeys<T extends Record<string, unknown>>(obj: T, keysToRemove: string[]): T {
+  const result = { ...obj };
+  keysToRemove.forEach((path) => {
+    const parts = path.split('.');
+    let current: Record<string, unknown> = result;
+    for (let i = 0; i < parts.length - 1; i++) {
+      const part = parts[i];
+      if (current[part] === undefined || current[part] === null) {
+        return; // Path doesn't exist, nothing to delete
+      }
+      current = current[part] as Record<string, unknown>;
+    }
+    const lastPart = parts[parts.length - 1];
+    if (current && lastPart in current) {
+      delete current[lastPart];
+    }
+  });
+  return result;
+}
+
+const getProjectOutputsPath = (projectRoot: string) => path.join(projectRoot, 'amplify_outputs.json');
+
+const getProjectOutputs = (projectRoot: string): $TSAny => {
+  const metaFilePath: string = getProjectOutputsPath(projectRoot);
+  return JSON.parse(fs.readFileSync(metaFilePath, 'utf8'));
+};
+
+export async function migrateCodegen(projRoot: string) {
+  await assert.doesNotReject(runCodegenCommand(projRoot), 'Codegen failed');
+}
+
+export async function deployGen2Sandbox(projRoot: string) {
+  await assert.doesNotReject(runGen2SandboxCommand(projRoot), 'Gen2 CDK deployment failed');
+}
```

Comment:
```suggestion
export async function assertMigrateCodegen(projRoot: string) {
  await assert.doesNotReject(runCodegenCommand(projRoot), 'Codegen failed');
}

export async function assertDeploysGen2Sandbox(projRoot: string) {
  await assert.doesNotReject(runGen2SandboxCommand(projRoot), 'Gen2 CDK deployment failed');
}
```

Or just get rid of these functions and have `run..` and assertions inline in relevant tests. (I'd just inline this, these functions seem too small).